packages/core/test/render3/component_spec.ts

```diff
@@ -703,4 +703,36 @@ describe('view destruction', () => {
             true,
             'Expected component onDestroy method to be called when its parent view is destroyed');
   });
+
+  it('should invoke onDestroy only once when a component is registered as a provider', () => {
+    const testToken = new InjectionToken<ComponentWithOnDestroy>('testToken');
+    let destroyCalls = 0;
+
+    class ComponentWithOnDestroy {
+      static ngComponentDef = defineComponent({
+        selectors: [['comp-with-on-destroy']],
+        type: ComponentWithOnDestroy,
+        consts: 0,
+        vars: 0,
+        factory: () => new ComponentWithOnDestroy(),
```

Comment:
The `directiveInject` in the template below should be moved here (this is what the compiler actually generates)

```suggestion
        factory: () => new ComponentWithOnDestroy(directiveInject(testToken),
```

Reply:
It can't be moved there, because it generates a circular dependency (`ComponentWithOnDestroy` is the one being provided as the `testToken`).